types/connect/index.d.ts

```diff
@@ -17,9 +18,11 @@ declare function createServer(): createServer.Server;
 declare namespace createServer {
     export type ServerHandle = HandleFunction | http.Server;
 
+    type NextFunction = (err?: any) => void;
+
     export type SimpleHandleFunction = (req: http.IncomingMessage, res: http.ServerResponse) => void;
-    export type NextHandleFunction = (req: http.IncomingMessage, res: http.ServerResponse, next: Function) => void;
-    export type ErrorHandleFunction = (err: Error, req: http.IncomingMessage, res: http.ServerResponse, next: Function) => void;
+    export type NextHandleFunction = (req: http.IncomingMessage, res: http.ServerResponse, next: NextFunction) => void;
+    export type ErrorHandleFunction = (err: any, req: http.IncomingMessage, res: http.ServerResponse, next: NextFunction) => void;
     export type HandleFunction = SimpleHandleFunction | NextHandleFunction | ErrorHandleFunction;
```

Comment:
@EvanHahn I think that `req` in the callbacks should use a subtype of `http.IncomingMessage` which also has `req.originalUrl: string`.
> In order to make middleware easier to write to be agnostic of the route, when the fn is invoked, the req.url will be altered to remove the route part (and the original will be available as req.originalUrl). For example, if fn is used at the route /foo, the request for /foo/bar will invoke fn with req.url === '/bar' and req.originalUrl === '/foo/bar'.
>-- https://www.npmjs.com/package/connect#appuseroute-fn

Reply:
That looks right to me. Would you be able to make a pull request? I don't think I'll have time, but could review it.

Reply:
I've never done pull requests before... But I'll try to do it later.

Reply:
Feel free to email me at <me@evanhahn.com> if I can help!